src/sentry/search/eap/columns.py

```diff
@@ -0,0 +1,67 @@
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any, Literal
+
+from sentry_protos.snuba.v1.trace_item_attribute_pb2 import AttributeAggregation, AttributeKey
+
+from sentry.search.eap import constants
+
+
+@dataclass(frozen=True)
+class ResolvedColumn:
+    # The alias for this column
+    public_alias: str  # `p95() as foo` -> `foo` or `p95()` -> `p95()`
+    # The internal rpc alias for this column
+    rpc_name: str
+    # The public type for this column
+    search_type: Literal["string", "number", "duration"]
+    # Processor is the function run in the post process step to transform data into the final result
+    processor: Callable[[Any], Any] | None = None
+    # Validator to check if the value in a query is correct
+    validator: Callable[[Any], None] | None = None
```

Comment:
Is the intention for the validator to raise an exception?

Reply:
Good q, I attempted to add one in the next PR. I think I'll change the type so it returns a bool so there's a default error message, but validators can raise their own error if they want